### PR TITLE
Add greedy labels resolver

### PR DESCRIPTION
**Pull request by sovrasov**
### Summary

This PR adds a simplified algorithm for resolving labels in hierarchical classification.

### Checklist

- [x] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [x] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Review by sungmanc — DISMISSED** [review]

File: otx/api/entities/label_schema.py [hunk 1]
```diff
@@ -558,6 +558,68 @@ def from_labels(cls, labels: Sequence[LabelEntity]) -> "LabelSchemaEntity":
         label_group = LabelGroup(name="from_label_list", labels=labels)
         return LabelSchemaEntity(label_groups=[label_group])
 
+    def resolve_labels_greedy(self, scored_labels: List[ScoredLabel]) -> List[ScoredLabel]:
```
Comment (Wovchena) [inline otx/api/entities/label_schema.py]:
```suggestion
    def resolve_labels_greedily(self, scored_labels: List[ScoredLabel]) -> List[ScoredLabel]:
```
You need an adverb, not adjective

File: otx/api/entities/label_schema.py [hunk 2]
```diff
@@ -558,6 +558,68 @@ def from_labels(cls, labels: Sequence[LabelEntity]) -> "LabelSchemaEntity":
...
+        """Resolves hierarchical labels and exclusivity based on a list of ScoredLabels (labels with probability).
+
+        The following two steps are taken:
+
+        - selects the most likely label from each label group
```
Comment (Wovchena) [inline otx/api/entities/label_schema.py]:
```suggestion
        - select the most likely label from each label group
```
After that you have
> \- add ...

without `s`

**Review by Wovchena — DISMISSED** [review]

**Review by AlexKoff88 — DISMISSED** [review]

File: otx/api/entities/label_schema.py [hunk 3]
```diff
@@ -558,6 +558,68 @@ def from_labels(cls, labels: Sequence[LabelEntity]) -> "LabelSchemaEntity":
         label_group = LabelGroup(name="from_label_list", labels=labels)
         return LabelSchemaEntity(label_groups=[label_group])
 
+    def resolve_labels_greedily(self, scored_labels: List[ScoredLabel]) -> List[ScoredLabel]:
+        """Resolves hierarchical labels and exclusivity based on a list of ScoredLabels (labels with probability).
+
+        The following two steps are taken:
+
+        - select the most likely label from each label group
+        - add their predecessors if they are also most likely labels (greedy approach).
```
Comment (AlexKoff88) [inline otx/api/entities/label_schema.py]: I would also add: "skip the candidate label if it is a successor of the label from another group that has a low probability"

**Review by AlexKoff88 — DISMISSED** [review]

File: otx/api/entities/label_schema.py (hunk 3 above)
Comment (sovrasov) [reply otx/api/entities/label_schema.py]: I'd say that this statement follows from the previous one, but my last formulation was a bit unclear, so I update it.

**Review by AlexKoff88 — APPROVED** [review]

**Comment by sovrasov** [comment]
@goodsong81 @sungmanc could you have a look?

**Review by sungmanc — APPROVED** [review]
LGTM

**Review by goodsong81 — APPROVED** [review]

**Pull request merged**